### ISSUE TITLE
Only sync modified or new files after a customizable delay of no changes.

### ISSUE THREAD

**Issue by mikeage** [nextcloud/desktop#6203]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you want to have the same feature implemented.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Feature description
I'm currently running Nextcloud on an Intel Mac, and one of the files I have in a backup directory is a Quicken QDF file. (I'm running Quicken under Codeweavers' Crossover (aka Wine), so it's possible that this behavior is due to the emulation). Quicken seems to regularly open and close the file, and it gets synced frequently via NC while I'm using the application. Frequently the sync fails as the file is modified while it's being uploaded, but even if not, it puts a significant load on my upload.

It'd be nice if there was a way to tell Nextcloud to only sync a file if it hasn't been modified in the last X seconds. In other words, build a list of files to sync, and then filter any file that was "recently" changed. If the file gets changed again, this will save time; if it doesn't, it'll get synced after a few seconds/minutes anyway, and so practically nothing is lost.

I don't know how new files are detected; if it's something like inotify or looking for a recent modification time or something else, so while I'm sure the "skip this file now" part is easy, I'm not sure how feasible it is to catch it on the next scan (or if there even are scans, or if's all on-demand once the client is running); I'm opening this to open discussion, not to demand an implementation.

**Comment by Mika-6-9** [comment]
Hello.
For me, this behavior you want is not suitable for multiple use in a team. This works for use in data saving mode if you are alone. This could be an option: work alone or in a team with different types of synchronization.
BUT Nextcloud is a collaborative tool and requires the most responsive mode of operation possible.
For our team use, we do not want to slow down the synchronization process.
Greetings

**Comment by mikeage** [comment]
> Hello.
> For me, this behavior you want is not suitable for multiple use in a team. This works for use in data saving mode if you are alone. This could be an option: work alone or in a team with different types of synchronization.
> BUT Nextcloud is a collaborative tool and requires the most responsive mode of operation possible.
> For our team use, we do not want to slow down the synchronization process.
> Greetings

I agree; the default behavior should not be changed. The proposal here is for a customizable parameter.

**Comment by js-d-coder** [comment]
I write notes on my Mac and I frequently get sync error as the notes app is updating the file every second. This feature will be very helpful.

**Comment by picarica** [comment]
1uping this, as someone who has firefox profile synced, everytime session changes closing or opening tab it syncs,  wopuld be very eneded

**Comment by Rello** [comment]
Hello,

the use case is a little limited. I also cannot find something like this on e.g. OneDrive.
Artificially delaying syncing in a sync client is not the purpose. Session files should be excluded from the sync then.
Also with the inevitable move to VFS, the systems will automatically take care of this with even less interaction by the user

**Comment by mikeage** [comment]
ok, thank you. You are correct that OneDrive doesn't have it... which is one reason that I don't like OneDrive ;-)

Note that it's not about avoiding syncing temporary files; that, as you mention, would be a case of exclusions. Instead it was about hysteresis, to avoid thrashing. And in an ideal world it wouldn't be necessary, because networks and remote file systems would be fast enough. But for now, I see it as something similar to a write cache, which also wouldn't need to exist if you could just write fast enough.
